### ISSUE TITLE
catch exception and skip faulty record

### ISSUE THREAD

**Issue by jorol** [gbv/PICA-Data#16]
PICA::Parser::Plus `croaks` if there's a faulty record in the dataset. Is there a way to catch this exception to skip faulty records within a `while` condition like

``` perl
while ( my $record = $parser->next ) {
    # process record
}
```

?

Thanks

**Comment by nichtich** [comment]
Well "faulty record" can be anything. In PICA::Parser::Plus and PICA::Parser::Plain the error is thrown on a faulty field. Should the field be skipped to the entire record? To enable such feature how about an option, such as

~~~perl
$parser = pica_parser( $type, skip_faulty_fields => 1)
while ( my $record = $parser->next ) { ... }
~~~

**Comment by jorol** [comment]
I would prefer to skip faulty fields by default and just throw a warning. What do you think?

**Comment by nichtich** [comment]
Ok, I favour an option `strict` to enable the current behaviour and skip faulty fields by default.

**Comment by jorol** [comment]
fixed in pr #18
